src/binary/ibc-setup/utils/resolve-option.ts

```diff
@@ -0,0 +1,14 @@
+export function resolveOption(
+  identifier: string,
+  ...args: Array<(string | undefined) | (() => string | undefined | void)>
```

Comment:
Can we remove `| void` here? The function should return something

Reply:
It is intended for usages like:

```diff
function getSomething() {
  if (someCondition) {
    return something
  }

--   return undefined
}
```

so you don't have to return `undefined`. But maybe it's better to be explicit. What do you think?

Reply:
My feel was that it would allow `() => {}` which returns void, but totally not acceptable. But I see your point.

Let's leave it, and only tighten it up if we shoot ourselves in the foot later. That is all types are in the end, guardrails to keep us safe when coding later.